### PR TITLE
Update to latest Timely

### PR DESCRIPTION
**Pull request by antiguru**
Bring in changes related to abomination and reference-counted addresses.

Signed-off-by: Moritz Hoffmann <antiguru@gmail.com>

### CODE REVIEW

File: src/trace/implementations/ord_neu.rs [hunk 1]
```diff
@@ -159,7 +159,6 @@ mod val_batch {
     ///
     /// The `L` parameter captures how the updates should be laid out, and `C` determines which
     /// merge batcher to select.
-    #[derive(Abomonation)]
```
Comment (frankmcsherry) [inline src/trace/implementations/ord_neu.rs:162]: Does this need a `#[derive(Serialize, Deserialize)]`? I don't know, and maybe we wont until we start to use it? :)

File: src/trace/implementations/ord_neu.rs [hunk 2]
```diff
@@ -720,7 +719,6 @@ mod key_batch {
...
```
Comment (frankmcsherry) [inline src/trace/implementations/ord_neu.rs:723]: Same as above; removing a derived `Abomonation` without introducing `Serialize, Deserialize`. Idk the consequences.

File: src/trace/implementations/rhh.rs [hunk 3]
```diff
@@ -250,8 +250,7 @@ mod val_batch {
     ///
     /// The `L` parameter captures how the updates should be laid out, and `C` determines which
     /// merge batcher to select.
-    #[derive(Abomonation)]
-    pub struct RhhValBatch<L: Layout> 
+    pub struct RhhValBatch<L: Layout>
```
Comment (frankmcsherry) [inline src/trace/implementations/rhh.rs:253]: Same Q as elsewhere.

**Review by frankmcsherry — APPROVED** [review]
All looks good to me, except a few moments where we lost an `Abomonation` derive and did not replace it with a `Serialize, Deserialize` derive. That might be fine (perhaps it was unused?), but I wanted to flag them in case they are known oopsies.

File: src/trace/implementations/ord_neu.rs (hunk 1 above)
Comment (antiguru) [reply src/trace/implementations/ord_neu.rs:162]: We cannot easily. Serde requires the GAT to be serialize/deserialize unless we tell it otherwise.

Comment (frankmcsherry) [reply src/trace/implementations/ord_neu.rs:162]: Gotcha. And .. no particular reason to want to serialize / deserialize these, outside perhaps of interest in some DD-based persistence? I guess I forget why the `derive(Abomonation)` was there in the first place, and perhaps there was no reason anymore?

**Pull request merged**